src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureSettingsForJsonNet.cs

```diff
@@ -0,0 +1,260 @@
+                                        }
+                                    }
+                                },
+                                OperationKind.Invocation);
+
+                            operationBlockStartAnalysisContext.RegisterOperationAction(
+                                (OperationAnalysisContext operationAnalysisContext) =>
+                                {
+                                    IObjectCreationOperation objectCreationOperation =
+                                        (IObjectCreationOperation)operationAnalysisContext.Operation;
+                                    if (jsonSerializerSettingsSymbol.Equals(objectCreationOperation.Type))
+                                    {
+                                        lock (rootOperationsNeedingAnalysis)
+                                        {
+                                            rootOperationsNeedingAnalysis.Add(
+                                                (objectCreationOperation.GetInnermostFunction(), operationAnalysisContext.ContainingSymbol));
+                                        }
+                                    }
+                                },
+                                OperationKind.ObjectCreation);
+
+                            operationBlockStartAnalysisContext.RegisterOperationAction(
+                                (OperationAnalysisContext operationAnalysisContext) =>
+                                {
+                                    IReturnOperation returnOperation = (IReturnOperation)operationAnalysisContext.Operation;
+                                    if (jsonSerializerSettingsSymbol.Equals(returnOperation.Type))
+                                    {
+                                        rootOperationsNeedingAnalysis.Add(
+                                            (returnOperation.GetInnermostFunction(), operationAnalysisContext.ContainingSymbol));
+                                    }
+                                },
+                                OperationKind.Return);
+                        });
+
+                    compilationStartAnalysisContext.RegisterCompilationEndAction(
+                        (CompilationAnalysisContext compilationAnalysisContext) =>
+                        {
+                            PooledDictionary<(Location Location, IMethodSymbol Method), HazardousUsageEvaluationResult> allResults = null;
+                            try
+                            {
+                                lock (rootOperationsNeedingAnalysis)
+                                {
+                                    if (!rootOperationsNeedingAnalysis.Any())
+                                    {
+                                        return;
+                                    }
+
+                                    allResults = PropertySetAnalysis.BatchGetOrComputeHazardousUsages(
```

Comment:
I think you might have explained this to me before, but why are we batching the computation at the of compilation instead of performing the analysis in each operation callback?

Reply:
Two (or more) PropertySetAnalysis DFA invocations may both have results for the same operation.  So want to merge all invocations' results together and only look at one result for a given operation.

Reply:
Hmm I am thinking about the perf impact. Each operation holds onto the semantic model, so you will basically hold onto all the semantic models for the root operations for the whole compilation lifetime.

Reply:
We can get back to the perf impact later, if measurements shows the same.